VirtoCommerce.CustomerModule.Data/Handlers/LogChangesMemberChangedEventHandler.cs

```diff
@@ -0,0 +1,249 @@
+            Member member, IEnumerable<string> originalEmails, IEnumerable<string> modifiedEmails)
+        {
+            return GetListChanges(
+                originalEmails,
+                modifiedEmails,
+                result =>
+                    (state, source, target) =>
+                    {
+                        if (state == EntryState.Added)
+                        {
+                            AddEmailAddedOperation(result, member, target);
+                        }
+                        else if (state == EntryState.Deleted)
+                        {
+                            AddEmailRemovedOperation(result, member, target);
+                        }
+                    });
+        }
+
+        protected virtual IEnumerable<string> GetAddressChanges(
+            Member member, IEnumerable<Address> originalAddress, IEnumerable<Address> modifiedAddress)
+        {
+            return GetListChanges(
+                originalAddress,
+                modifiedAddress,
+                result =>
+                    (state, source, target) =>
+                    {
+                        if (state == EntryState.Added)
+                        {
+                            AddAddressAddedOperation(result, member, target);
+                        }
+                        else if (state == EntryState.Deleted)
+                        {
+                            AddAddressRemovedOperation(result, member, target);
+                        }
+                    });
+        }
+
+        protected virtual IEnumerable<string> GetListChanges<T>(
+            IEnumerable<T> originalCollection,
+            IEnumerable<T> modifiedCollection,
+            Func<List<string>, Action<EntryState, T, T>> compareAction)
+        {
+            if (originalCollection == null || modifiedCollection == null)
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var result = new List<string>();
+            modifiedCollection
+                .Where(x => x != null).ToList()
+                .CompareTo(
+                    originalCollection.Where(x => x != null).ToList(),
+                    EqualityComparer<T>.Default,
+                    compareAction(result));
+            return result;
+        }
+
+        private static void AddMemberPropertyChangedOperation(
+            List<string> result, GenericChangedEntry<Member> changedEntry, Difference difference)
+        {
+            result.Add(
+                string.Format(
+                    MemberResources.MemberPropertyChanged,
+                    changedEntry.OldEntry.MemberType,
+                    changedEntry.NewEntry.Name,
+                    difference.Name,
+                    difference.OldValue,
+                    difference.NewValue));
+        }
+
+        private void AddPhoneRemovedOperation(List<string> result, Member member, string phone)
```

Comment:
Can we call these methods directly in-place instead of surround them by special methods?

Reply:
There are 30 lines of code which do nothing.